Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -316,6 +319,89 @@ WarpX::MacroscopicEvolveE (int lev, PatchType patch_type, amrex::Real a_dt) {
     }
 }
 
+void
+WarpX::DampFieldsInGuards(std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield,
+                          std::array<std::unique_ptr<amrex::MultiFab>,3>& Bfield) {
+
+    for ( amrex::MFIter mfi(*Efield[0], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi )
+    {
+
+        amrex::Array4<amrex::Real> const& Ex_arr = Efield[0]->array(mfi);
+        amrex::Array4<amrex::Real> const& Ey_arr = Efield[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& Ez_arr = Efield[2]->array(mfi);
+        amrex::Array4<amrex::Real> const& Bx_arr = Bfield[0]->array(mfi);
+        amrex::Array4<amrex::Real> const& By_arr = Bfield[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& Bz_arr = Bfield[2]->array(mfi);
+
+        // Get the tilebox from Efield so that it includes the guard cells.
+        amrex::Box tilebox = (*Efield[0])[mfi].box();
+        int const nz_tile = tilebox.bigEnd(AMREX_SPACEDIM-1);
```

Comment:
Since `AMREX_SPACEDIM-1` is used a few times here, I wonder if it could be cleaner to define an auxiliary integer variable `const int zdir = AMREX_SPACEDIM-1;` and use it everywhere as the index of the z direction.

Reply:
Yes, I can do this. This is done elsewhere. The change is pushed.